### PR TITLE
fix: No warning if the number of elements in the spread operator using the if statement is more than two.

### LINKED ISSUE #68: Even though it contains multiple elements, the avoid_single_child warning appears when using an if condition.
**Describe the bug**
Even though it contains multiple elements, the avoid_single_child warning appears when using an if condition.

**To Reproduce**
```dart
Column(      // <- avoid_single_child warnings here
  children: [
    if (isA) ...[
      _buildA1(),
      _buildA2(),
    ] else if (isB) ...[
      _buildB1(),
      _buildB2(),
    ] else ...[
      _buildC1(),
      _buildC2(),
    ],
  ],
)
```

**Expected behavior**
if condition children has not warnings

### PR DESCRIPTION
**Pull request by boywithdv**
## 🔗 Related Issues

- closes #68 

## 🙌 What's Done

- Do not display a warning if the if statement written to a child element has two or more elements with the spread operator

## ✍️ What's Not Done

- 

## 🖼️ Image Differences

| not display warnings  | warning     |
| --------- | --------- |
| スクリーンショット 2024-11-05 16 56 34 | スクリーンショット 2024-11-05 16 56 47 |

Both if and else will similarly display a warning message if the number of elements is one.

## 🤼 Desired Review Method

- [x] Correction Commit
- [ ] Pair programming

> [!NOTE]
> It is possible that a reviewer's will may cause a method to be implemented that is not selected.

## Pre-launch Checklist
- [x] I have reviewed my own code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I updated/added relevant documentation (doc comments with ///).

### CODE REVIEW

**Review by riscait — COMMENTED** [review]
badge
@boywithdv 

The issues in the corresponding Issue are as follows:

> Even though it contains multiple elements, the avoid_single_child warning appears when using an if condition.

Looking at the PR it looks like another revision, what is the intent?

**Comment by boywithdv** [comment]
> Looking at the PR it looks like another revision, what is the intent?


The title and summary description of the PR did not seem to match the revisions.

We have just corrected the title and summary description.

The PR has been modified to display a warning when the number of elements in the if statement described in the child element is two or more, and not when the number is one.

File: packages/altive_lints/example/lints/avoid_single_child.dart [hunk 1]
```diff
@@ -70,14 +70,47 @@ class MyWidget extends StatelessWidget {
...
+        ),
         Column(
           children: random.nextBool()
               ? [
                   const Text('Hello World'),
                 ]
               : [
                   const Text('Hello'),
-                  const Text('World'),
                 ],
```
Comment (riscait) [inline packages/altive_lints/example/lints/avoid_single_child.dart]: badge
This is not the subject of this Issue, but this pattern has one element each, so lint is the expected value.

File: packages/altive_lints/lib/src/lints/avoid_single_child.dart [hunk 2]
```diff
@@ -85,4 +96,12 @@ class AvoidSingleChild extends DartLintRule {
       }
     });
   }
+
+  bool _hasSingleChild(CollectionElement element) {
+    if (element is SpreadElement && element.expression is ListLiteral) {
+      final spreadElement = element.expression as ListLiteral;
+      return spreadElement.elements.length != 1;
+    }
```
Comment (riscait) [inline packages/altive_lints/lib/src/lints/avoid_single_child.dart]: badge
@boywithdv 
What does the name `_hasSingleChild` mean when `length != 1`, what does the name `_hasSingleChild` mean?It seems intuitive if `length == 1`...

Comment (boywithdv) [reply packages/altive_lints/lib/src/lints/avoid_single_child.dart]: Indeed, at a quick glance, it seems that the function name does not match the actual situation.

The function name _hasSingleChild seems to return true in the case of a single element, so the function name should be changed to _hasMultipleChild to make it clear that it returns true when there are multiple child elements.

File: packages/altive_lints/lib/src/lints/avoid_single_child.dart [hunk 3]
```diff
@@ -85,4 +96,12 @@ class AvoidSingleChild extends DartLintRule {
       }
     });
   }
+
+  bool _hasMultipleChild(CollectionElement element) {
+    if (element is SpreadElement && element.expression is ListLiteral) {
+      final spreadElement = element.expression as ListLiteral;
+      return spreadElement.elements.length != 1;
```
Comment (riscait) [inline packages/altive_lints/lib/src/lints/avoid_single_child.dart]: badge
@boywithdv 
Is there a problem if `isEmpty` is true, as it seems to `return true` as well?
```suggestion
      return spreadElement.elements.length > 1;
```

Comment (boywithdv) [reply packages/altive_lints/lib/src/lints/avoid_single_child.dart]: It seems more natural to set it to 1 or more, since it returns true even if it is empty.

I was not able to assume that the number of elements in the spread operator is empty.

We will also create a test case for the case where the number of elements of the spread operator is empty.
Thank you very much.

File: packages/altive_lints/lib/src/lints/avoid_single_child.dart [hunk 4]
```diff
@@ -76,7 +76,18 @@ class AvoidSingleChild extends DartLintRule {
...
           return;
         }
+        for (final element in childrenList.elements) {
+          if (element is IfElement) {
+            if (_hasMultipleChild(element.thenElement)) {
+              return;
+            }
 
+            if (element.elseElement != null &&
+                _hasMultipleChild(element.elseElement!)) {
```
Comment (riscait) [inline packages/altive_lints/lib/src/lints/avoid_single_child.dart]: badge
`! ` as little as possible. Do you know why?

```suggestion
            if (element.elseElement case final CollectionElement ce
                when _hasMultipleChild(ce)) {
```

File: packages/altive_lints/lib/src/lints/avoid_single_child.dart (hunk 2 above)
Comment (riscait) [reply packages/altive_lints/lib/src/lints/avoid_single_child.dart]: Names are important. Make sure you name it with proper thought and self-review to make sure it is correct.

**Review by riscait — APPROVED** [review]
Thanks for adding the test!
I commented on some of them, so please check them and merge them if they are OK.

File: packages/altive_lints/lib/src/lints/avoid_single_child.dart (hunk 4 above)
Comment (boywithdv) [reply packages/altive_lints/lib/src/lints/avoid_single_child.dart]: concise rather than long conditionals from a readability and maintainability standpoint! is to be avoided as much as possible.
Thank you.
